zmq.hpp

```diff
@@ -615,8 +620,340 @@ inline void swap(context_t &a, context_t &b) ZMQ_NOTHROW {
     a.swap(b);
 }
 
+#ifdef ZMQ_CPP11
+struct send_result
+{
+    size_t size;    // message size in bytes
+    bool success;
+};
+
+struct recv_result
+{
+    size_t size;    // message size in bytes
+    bool success;
+};
+
+struct recv_buffer_result
+{
+    size_t size;    // number of bytes written to buffer
+    size_t untruncated_size;  // untruncated message size in bytes
+    bool success;
+
+    ZMQ_NODISCARD bool truncated() const noexcept
+    {
+        return size != untruncated_size;
+    }
+};
+
+enum class send_flags : int
+{
+    none = 0,
+    dontwait = ZMQ_DONTWAIT,
+    sndmore = ZMQ_SNDMORE
+};
+
+constexpr send_flags operator|(send_flags a, send_flags b) noexcept
+{
+    return static_cast<send_flags>(static_cast<int>(a) | static_cast<int>(b));
```

Comment:
Using std::underlying_type here would reduce redundancy (but would be more verbose):
```suggestion
    return static_cast<send_flags>(static_cast<std::underlying_type<send_flags>::type>(a)
        | static_cast<<std::underlying_type<send_flags>::type>(b));
```